src/app/common/audio-player/audio-player.coffee

```diff
@@ -0,0 +1,132 @@
+angular.module('doubtfire.common.audio-player', [])
+
+.directive 'audioPlayer', ->
+  restrict: 'E'
+  replace: true
+  templateUrl: 'common/audio-player/audio-player.tpl.html'
+
+  controller: ($scope, CommentResourceService, Task, $sce, $http, $q) ->
+    DEFAULT_HEIGHT = 300
+
+    $scope.loadPlayer = false
+
+    $scope.commentAudioUrl = CommentResourceService.commentAudioUrl
+
+    #=========================================================================
+    getAudioComment = (comment, audioElement) ->
+      audioUrl = $sce.trustAsResourceUrl(Task.generateCommentsAttachmentUrl($scope.project, $scope.task, comment))
+      $http({url: audioUrl, method: "get", withCredentials: true, params: ""}, responseType: "text").then (response) ->
+        audioElement.src = $sce.trustAsResourceUrl(response.data)
+
+    #=========================================================================
+    # AUDIO PLAYER SETUP
+    $scope.audioPlayerSetup = (index, comment)->
+
+      isPlaying = false
+      btnPlay = document.getElementById('btnPlay-' + index)
+      audioElement = document.getElementById('audio-element-' + index)
+      getAudioComment(comment,audioElement)
+      audioElement.crossOrigin = "anonymous"
+      canvasElement = document.getElementById('audio-visualiser-' + index)
+
+      CommentResourceService.setupAudioContext()
+      audioCtx = CommentResourceService.audioContext
```

Comment:
Keep this aspect of CommentResourceService - keeping track of the context and testing for capability/support